includes/collection/class-inbox.php

```diff
@@ -27,32 +27,72 @@ class Inbox {
 	 */
 	const POST_TYPE = 'ap_inbox';
 
+	/**
+	 * Context for user inbox requests.
+	 *
+	 * @var string
+	 */
+	const CONTEXT_INBOX = 'inbox';
+
+	/**
+	 * Context for shared inbox requests.
+	 *
+	 * @var string
+	 */
+	const CONTEXT_SHARED_INBOX = 'shared_inbox';
+
 	/**
 	 * Add an activity to the inbox.
 	 *
-	 * @param Activity|\WP_Error $activity The Activity object.
-	 * @param int                $user_id  The id of the local blog-user.
+	 * @param Activity|\WP_Error $activity   The Activity object.
+	 * @param int|array          $recipients The id(s) of the local blog-user(s).
+	 *                                       Accepts single int for backward compatibility.
 	 *
 	 * @return false|int|\WP_Error The added item or an error.
 	 */
-	public static function add( $activity, $user_id ) {
+	public static function add( $activity, $recipients ) {
 		if ( \is_wp_error( $activity ) ) {
 			return $activity;
 		}
 
-		$item = self::get_by_guid( $activity->get_id() );
+		// Backward compatibility: convert single user_id to array.
+		if ( ! \is_array( $recipients ) ) {
+			$recipients = array( (int) $recipients );
+		}
+
+		// Sanitize recipients.
+		$recipients = \array_map( 'absint', $recipients );
+		$recipients = \array_unique( $recipients );
+		// Filter out null/false but keep 0 (blog user).
+		$recipients = \array_filter(
+			$recipients,
+			function ( $value ) {
+				return null !== $value && false !== $value;
+			}
+		);
```

Comment:
Is it possible to still have `null` and `false` values after `absint()`? We might be able to just remove this.